src/metadata/TerraformMetadata.js

```diff
@@ -1,8 +1,207 @@
+      component.definedAttributes
+        .forEach((attribute) => this.getLinkDefinition(links, component.type, attribute));
+    });
+    return links;
+  }
+
+  /**
+   * Set link definition from attribute in link definition array.
+   * @param {ComponentLinkDefinition[]} links - Link definition array.
+   * @param {String} componentType - Type of related component.
+   * @param {ComponentAttributeDefinition} definedAttribute - Attribute to parse.
+   */
+  getLinkDefinition(array, componentType, definedAttribute) {
+    if (definedAttribute.definedAttributes.length > 0) {
+      definedAttribute.definedAttributes
+        .forEach((attribute) => this.getLinkDefinition(array, componentType, attribute));
+    }
+
+    if (definedAttribute.type !== 'Link') {
+      return;
+    }
+
+    const linkDefinition = new ComponentLinkDefinition({ type: definedAttribute.linkType });
+    if (definedAttribute.linkType === 'Reverse') {
+      linkDefinition.sourceRef = definedAttribute.linkRef;
+      linkDefinition.targetRef = componentType;
+    } else {
+      linkDefinition.sourceRef = componentType;
+      linkDefinition.targetRef = definedAttribute.linkRef;
+    }
+
+    array.push(linkDefinition);
+  }
+
+  /**
+   * Get all component definitions from metadata.
+   * @return {ComponentDefinition[]} - Array of component definitions.
+   */
+  getComponentDefinitions() {
+    const definitions = [];
+    Object.keys(this.providers)
+      .map((key) => this.providers[key])
+      .forEach((provider) => {
+        definitions.push(this.getProviderDefinition(provider));
+
+        provider.data.forEach((data) => definitions.push(
+          this.getBlockDefinition('data', provider.name, data),
+        ));
+
+        provider.modules.forEach((module) => definitions.push(
+          this.getBlockDefinition('module', provider.name, module),
+        ));
+
+        provider.resources.forEach((resource) => definitions.push(
+          this.getBlockDefinition('resource', provider.name, resource),
+        ));
+
+        provider.variables.forEach((variable) => definitions.push(
+          this.getBlockDefinition('variable', provider.name, variable),
+        ));
+      });
+    return definitions;
+  }
+
+  /**
+   * Get provider definition.
+   * @param {Object} provider - Provider from metadata to parse.
+   * @returns {TerraformComponentDefinition} Parsed provider component Definition.
+   */
+  getProviderDefinition(provider) {
+    const definition = new TerraformComponentDefinition({
+      blockType: 'provider',
+      provider: provider.name,
+      type: provider.name,
+      icon: provider.icon,
+      model: provider.model,
+      definedAttributes: provider.attributes.map(this.getAttributeDefinition),
+      isContainer: provider.isContainer,
+    });
+
+    definition.parentTypes = this.getParentTypes(definition);
+
+    return definition;
+  }
+
+  /**
+   * Get block definition, can be module/data/resource/variable.
```

Comment:
`can be module/data/resource/variable.` repeated just under, so a bit useless

Reply:
done